src/transport/raw/tests/TestTCP.cpp

```diff
@@ -111,81 +182,182 @@ void CheckSimpleInitTest6(nlTestSuite * inSuite, void * inContext)
+    Inet::TCPEndPoint * lEndPoint       = tcp.FindActiveConnection(lPeerAddress);
+    NL_TEST_ASSERT(inSuite, lEndPoint != nullptr);
+
+    chip::System::PacketBufferHandle buffer;
+    CHIP_ERROR err = CHIP_NO_ERROR;
 
-// Test Suite
+    // Test a single packet buffer.
+    expected_lengths                                   = (const uint16_t[]){ 11, 0 };
+    buffer                                             = PreparePacketBuffers(inSuite, expected_lengths);
+    gMockTransportMgrDelegate.mReceiveHandlerCallCount = 0;
+    err                                                = tcp.ProcessReceivedBuffer(lEndPoint, lPeerAddress, std::move(buffer));
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, gMockTransportMgrDelegate.mReceiveHandlerCallCount == 1);
+
+    // Test a message in a chain of three packet buffers.
+    const uint16_t k2Lengths[]                         = { 21, 22, 23, 0 };
+    expected_lengths                                   = (const uint16_t[]){ 66, 0 };
+    buffer                                             = PreparePacketBuffers(inSuite, k2Lengths);
+    gMockTransportMgrDelegate.mReceiveHandlerCallCount = 0;
+    err                                                = tcp.ProcessReceivedBuffer(lEndPoint, lPeerAddress, std::move(buffer));
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, gMockTransportMgrDelegate.mReceiveHandlerCallCount == 1);
+
+    // Test two messages in a chain.
+    const uint16_t k3Lengths1[] = { 31, 0 };
+    const uint16_t k3Lengths2[] = { 32, 0 };
+    expected_lengths            = (const uint16_t[]){ 31, 32, 0 };
+    buffer                      = PreparePacketBuffers(inSuite, k3Lengths1);
+    buffer->AddToEnd(PreparePacketBuffers(inSuite, k3Lengths2));
+    gMockTransportMgrDelegate.mReceiveHandlerCallCount = 0;
+    err                                                = tcp.ProcessReceivedBuffer(lEndPoint, lPeerAddress, std::move(buffer));
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, gMockTransportMgrDelegate.mReceiveHandlerCallCount == 2);
+
+    // Test a chain of two messages, each a chain.
+    const uint16_t k4Lengths1[] = { 41, 42, 0 };
+    const uint16_t k4Lengths2[] = { 43, 44, 0 };
+    expected_lengths            = (const uint16_t[]){ 83, 87, 0 };
+    buffer                      = PreparePacketBuffers(inSuite, k4Lengths1);
+    buffer->AddToEnd(PreparePacketBuffers(inSuite, k4Lengths2));
+    gMockTransportMgrDelegate.mReceiveHandlerCallCount = 0;
+    err                                                = tcp.ProcessReceivedBuffer(lEndPoint, lPeerAddress, std::move(buffer));
+    NL_TEST_ASSERT(inSuite, err == CHIP_NO_ERROR);
+    NL_TEST_ASSERT(inSuite, gMockTransportMgrDelegate.mReceiveHandlerCallCount == 2);
+
+    // Test a chain that is too large to coalesce into a single packet buffer, followed by a normal message.
```

Comment:
How easy would it be to add a test for a TCP message that claims to be `UINT16_MAX` or `UINT16_MAX-1` bytes long?  Followup for doing that, maybe, if it's not trivial?

Reply:
I'll do this as part of #5438 in a prompt followup.